internal/keepers/filter.go

```diff
@@ -80,18 +81,16 @@ func (rc *reportCoordinator) Filter() func(types.UpkeepKey) bool {
 			// TODO: the key is constructed in the registry. splitting out the
 			// block number here is a hack solution that should be fixed asap.
 			// AUTO-1480
-			var blKey string
+			var blKey int
 			parts := strings.Split(string(key), "|")
 			if len(parts) == 2 {
-				blKey = parts[0]
+				blKey, _ = strconv.Atoi(parts[0])
 			}
 
-			// only apply filter if key block is after block in cache
-			if len(bl.TransmitBlockNumber) > 0 && blKey > string(bl.TransmitBlockNumber) {
-				return true
-			}
+			transmitBlockNumber, _ := strconv.Atoi(string(bl.TransmitBlockNumber))
```

Comment:
there's also a case when `bl.TransmitBlockNumber` can be empty string where we would have returned false, i think we've changed the logic in that case

Reply:
Done